src/components/common/dashboard/Table.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+import { Table, Thead, Tbody, Tr, Th, Td, Heading } from '@chakra-ui/react';
+
+const composeTableHeader = header => {
+  return <Th>{header}</Th>;
+};
+
+const composeTableBody = row => {
+  const { name, email, action } = row;
+  return (
+    <Tr>
+      <Td>
+        <Heading size="h4">{name}</Heading>
+      </Td>
+      <Td>{email}</Td>
+      <Td fontWeight="semibold" color="#C70E0E">
+        {action}
+      </Td>
+    </Tr>
+  );
+};
+
+const DataTable = props => {
+  const { headers, data } = props;
+  return (
+    <Table>
+      <Thead>
+        <Tr>{headers.map(composeTableHeader)}</Tr>
+      </Thead>
+      <Tbody>{data.map(composeTableBody)}</Tbody>
+    </Table>
+  );
+};
+
+export default DataTable;
```

Comment:
Thoughts on naming this component `SampleTable` and having it live somewhere under the `themes` folder. Folks would build their own tables using the components you styled — but write their own `composeTableHeader` and `composeTableBody` functions for each individual table depending on the data shape?

The reasoning is, you *might* be able to provide super generic `compose__` functions, but I think it will be very difficult - and once a table has a data shape that looks different, the functions don't work anymore. So I'm thinking of using this file as an example of how to make a table with the components you styled

This would also allow you to use `react-table` as well, you can explore an example with that library (which essentially provides these compose functions + some other stuff). No reason to rewrite logic that some library as already solved :)  See https://chakra-ui.com/guides/integrations/with-react-table